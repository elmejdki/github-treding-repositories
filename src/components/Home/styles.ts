import styled, { css } from 'styled-components';

const SIZING = css`
  max-width: 1400px;
  margin: 0 30px;

  @media screen and (min-width: 1400px) {
    margin: 0 auto;
  }
`;

export const Header = styled.header`
  font-size: 32px;
  font-weight: bold;
  text-align: center;
  padding: 50px 5px;
  color: #424242;

  ${SIZING}
`;

export const Main = styled.main`
  ${SIZING}
`;

export const Footer = styled.footer`
  font-weight: bold;
  text-align: center;
  padding: 20px;
  color: #a1a1a1;

  ${SIZING}
`;
