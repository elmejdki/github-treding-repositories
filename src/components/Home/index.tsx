import { Main, Header, Footer } from './styles';
import RepositoryListFilters from '../RepositoryListFilters';
import RepositoryList from '../RepositoryList';

const Home = () => {
  return (
    <>
      <Header>
        Trending GITHUB repositories in the last 7 days
      </Header>
      <Main>
        <RepositoryListFilters />
        <RepositoryList />
      </Main>
      <Footer>Â© Copyright Zakariae El Mejdki 2022.</Footer>
    </>
  );
};

export default Home;
