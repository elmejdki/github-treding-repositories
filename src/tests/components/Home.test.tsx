import * as ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import Home from '../../components/Home';

const renderer = ShallowRenderer.createRenderer();

it('should render Home correctly', () => {
  renderer.render(<Home />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
