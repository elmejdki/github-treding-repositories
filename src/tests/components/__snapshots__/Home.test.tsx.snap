// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home correctly 1`] = `
<React.Fragment>
  <styled.header>
    Trending GITHUB repositories in the last 7 days
  </styled.header>
  <styled.main>
    <RepositoryListFilters />
    <RepositoryList />
  </styled.main>
  <styled.footer>
    Â© Copyright Zakariae El Mejdki 2022.
  </styled.footer>
</React.Fragment>
`;
